doc/introduction/compiling_circuits.rst

```diff
@@ -373,6 +340,197 @@ From here, we can iterate through the stages of decomposition:
 ───H†─│────────────╰●───────────────┤  
 ──────╰●────────────────────────────┤  
 
+Custom Operator Decomposition
+-----------------------------
+
+When executing QNodes on a device, PennyLane will automatically decompose gates 
+that are unsupported by the device using built-in decomposition rules.
+
+In addition, you can provide *new* decomposition rules to be used, but the behaviour
+and user-interface is different depending on if the new decompositions system is
+enabled.
```

Comment:
```suggestion
and user-interface is different depending on if the graph decompositions system is
enabled.
```

I would err towards not calling it 'new decomposition' system as it relies on readers knowing the temporal order of our systems. Rather, better to refer to it based on its feature/algorithm ('graph decompositions'). This also matches the function for turning it on :)